realm/realm-library/src/androidTest/java/io/realm/RealmMigrationTests.java

```diff
@@ -544,6 +558,29 @@ public void migrate(DynamicRealm realm, long oldVersion, long newVersion) {
         }
     }
 
+    @Test(expected = FileNotFoundException.class)
+    public void migrateRealm_config_nonExistingRealmFile() throws FileNotFoundException {
+        RealmConfiguration config = configFactory.createConfigurationBuilder().migration(new RealmMigration() {
+            @Override
+            public void migrate(DynamicRealm realm, long oldVersion, long newVersion) {
+
+            }
+        }).build();
+        Realm.migrateRealm(config);
+    }
+
+    @Test(expected = FileNotFoundException.class)
```

Comment:
Don't use this annotation if the test case contains multiple lines of code. The wrong code might throw instead: https://github.com/realm/realm-java/blob/master/CONTRIBUTING.md#unit-tests ;)